### PR TITLE
Error early if apply test vectors returns INVALID payload

### PR DESCRIPTION
**Pull request by debjit-bw**

### CODE REVIEW

File: scripts/generate_test_vectors.sh [hunk 1]
```diff
@@ -24,14 +24,28 @@ until curl -X POST -H "Content-Type: application/json" \
     sleep 2
 done
 
-BLOCK_COUNTER=0
+echo "Nethermind is available"
```
Comment (dapplion) [inline scripts/generate_test_vectors.sh:27]: This script maybe called for various clients, it's not nethermind only. You can say "EL is available"

File: scripts/generate_test_vectors.sh [hunk 2]
```diff
@@ -24,14 +24,28 @@ until curl -X POST -H "Content-Type: application/json" \
...
+
+declare -i BLOCK_COUNTER=0
 
 function make_block() {
-  ((BLOCK_COUNTER++))
+  # increment block counter
+  BLOCK_COUNTER=$((BLOCK_COUNTER + 1))
+
+  echo "Making block $BLOCK_COUNTER"
```
Comment (dapplion) [inline scripts/generate_test_vectors.sh:35]: This appears to be a double log between this line and 163

File: scripts/generate_test_vectors.sh [hunk 3]
```diff
@@ -24,14 +24,28 @@ until curl -X POST -H "Content-Type: application/json" \
...
 
-BLOCK_COUNTER=0
+echo "Nethermind is available"
+
+declare -i BLOCK_COUNTER=0
 
 function make_block() {
-  ((BLOCK_COUNTER++))
+  # increment block counter
+  BLOCK_COUNTER=$((BLOCK_COUNTER + 1))
```
Comment (dapplion) [inline scripts/generate_test_vectors.sh:33]: What's the motivation to change this syntax?

File: scripts/generate_test_vectors.sh (hunk 1 above)
Comment (debjit-bw) [reply scripts/generate_test_vectors.sh]: done

File: scripts/generate_test_vectors.sh (hunk 2 above)
Comment (debjit-bw) [reply scripts/generate_test_vectors.sh:35]: removed from line 163

File: scripts/generate_test_vectors.sh (hunk 3 above)
Comment (debjit-bw) [reply scripts/generate_test_vectors.sh:33]: difference in how this syntax is handled between sh and bash. earlier was getting an error  where docker image was exiting as soon as the make_block function got called, [this commit](https://github.com/debjit-bw/reth_gnosis/commit/d2439585e95efeb2be693a6407e01fa4bffee832) fixed it

just verified it again with everything same, just the syntax reverted (https://github.com/debjit-bw/reth_gnosis/actions/runs/11368078320/job/31622284450)

**Review by dapplion — APPROVED** [review]

**Pull request merged**